test/expectations/expectations.go

```diff
@@ -83,6 +84,18 @@ func ExpectApplied(ctx context.Context, c client.Client, objects ...client.Objec
 	}
 }
 
+func ExpectStatusConditions(ctx context.Context, c client.Client, obj status.Object, conditions ...status.Condition) {
```

Comment:
How useful is this? Do we need to check exact values or can we just expect testers that are creating assertions to just validate the status of the condition?

Reply:
On top of this, someone could just write `Expect(conditions, HasElements(...))` rather than using this function

Reply:
You don't have to care about observedGeneration or LastTransitionTime, so doing hasElements doesn't work since it's not strict equality